integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/PlanUtils.java

```diff
@@ -219,20 +219,34 @@ public static OpenLineage.DatasourceDatasetFacet datasourceFacet(
    * and namespace.
    *
    * @param parentRunId
-   * @param parentJob
+   * @param parentJobName
    * @param parentJobNamespace
    * @return
    */
   public static OpenLineage.ParentRunFacet parentRunFacet(
-      UUID parentRunId, String parentJob, String parentJobNamespace) {
+      UUID parentRunId,
+      String parentJobName,
+      String parentJobNamespace,
+      UUID rootParentRunId,
+      String rootParentJobName,
+      String rootParentJobNamespace) {
     return new OpenLineage(Versions.OPEN_LINEAGE_PRODUCER_URI)
         .newParentRunFacetBuilder()
         .run(new OpenLineage.ParentRunFacetRunBuilder().runId(parentRunId).build())
         .job(
             new OpenLineage.ParentRunFacetJobBuilder()
-                .name(NameNormalizer.normalize(parentJob))
+                .name(NameNormalizer.normalize(parentJobName))
                 .namespace(parentJobNamespace)
                 .build())
+        .root(
+            new OpenLineage.ParentRunFacetRootBuilder()
+                .run(new OpenLineage.RootRunBuilder().runId(rootParentRunId).build())
+                .job(
+                    new OpenLineage.RootJobBuilder()
+                        .namespace(rootParentJobNamespace)
+                        .name(rootParentJobName)
+                        .build())
+                .build())
```

Comment:
IMHO `parent.root` facet filled up with null-only values looks weird. Maybe it should be set only if all fields are not-null.

Reply:
Those are not null - as for the semantics of those fields, they get filled by respective parent ids, then if not present, current job ids.

```
        eventEmitter
            .getRootParentRunId()
            .orElse(eventEmitter.getParentRunId().orElse(eventEmitter.getApplicationRunId())),
        eventEmitter
            .getRootParentJobName()
            .orElse(eventEmitter.getParentJobName().orElse(eventEmitter.getApplicationJobName())),
        eventEmitter
            .getRootParentJobNamespace()
            .orElse(eventEmitter.getParentJobNamespace().orElse(eventEmitter.getJobNamespace())));
```

Reply:
I see, thanks for clarification.